dataset/dataset.go

```diff
@@ -467,15 +476,63 @@ func (c *Client) GetVersions(ctx context.Context, userAuthToken, serviceAuthToke
 		return
 	}
 
-	versions := struct {
-		Items []Version `json:"items"`
-	}{}
+	if err = json.Unmarshal(b, &m); err != nil {
+		return
+	}
 
-	err = json.Unmarshal(b, &versions)
-	m = versions.Items
 	return
 }
 
+// GetVersionsInBatches retrieves a list of datasets in concurrent batches and accumulates the results
+func (c *Client) GetVersionsInBatches(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceAuthToken, collectionID, datasetID, edition string, batchSize, maxWorkers int) (versions VersionsList, err error) {
+
+	// Function to aggregate items.
+	// For the first received batch, as we have the total count information, will initialise the final structure of items with a fixed size equal to TotalCount.
+	// This serves two purposes:
+	//   - We can guarantee, even with concurrent calls, that values are returned in the same order that the API defines, by offsetting the index.
+	//   - We do a single memory allocation for the final array, making the code more memory efficient.
+	var processBatch VersionsBatchProcessor = func(b VersionsList) (abort bool, err error) {
+		if len(versions.Items) == 0 { // first batch response being handled
+			versions.TotalCount = b.TotalCount
+			versions.Items = make([]Version, b.TotalCount)
+			versions.Count = b.TotalCount
+		}
+		for i := 0; i < len(b.Items); i++ {
+			versions.Items[i+b.Offset] = b.Items[i]
+		}
+		return false, nil
```

Comment:
I'm not sure I follow the purpose of returning the `abort bool` - it's always returning `false`. So my question is is this needed?

Reply:
🤷 was c&p from other batch getters/processors

Reply:
I realised after post this comment this function definition is a pattern used by the batch processor so it's fine as is because this is the expected structure for an established existing function.